triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala

```diff
@@ -331,4 +325,52 @@ class ServiceTest extends AsyncFlatSpec with Eventually with Matchers with Postg
       ledgerProxy.enable()
     }
   }
+
+  it should "stop a failing trigger that can't be restarted" in withHttpService(Some(dar)) {
+    (uri: Uri, client: LedgerClient, ledgerProxy: Proxy) =>
+      // Simulate the ledger becoming unrecoverably unavailable due to
+      // network connectivity loss. The stop strategy means the running
+      // trigger will be terminated.
+      for {
+        // Request a trigger be started for Alice.
+        resp <- startTrigger(uri, s"$testPkgId:TestTrigger:trigger", "Alice")
+        aliceTrigger <- parseTriggerId(resp)
+        triggerId <- parseTriggerId(resp)
+        // Proceed when it's confirmed to be running.
+        _ <- assertTriggerIds(uri, "Alice", _ == Vector(aliceTrigger))
+        // Simulate unrecoverable network connectivity loss.
+        _ <- Future { ledgerProxy.disable() }
+        // Confirm that the running trigger is stopped.
+        _ <- assertTriggerIds(uri, "Alice", _.isEmpty)
+      } yield succeed
+  }
+
+  it should "restart a failing trigger if possible" in withHttpService(Some(dar)) {
+    (uri: Uri, client: LedgerClient, ledgerProxy: Proxy) =>
+      // Simulate the ledger briefly being unavailable due to network
+      // connectivity loss. Our restart strategy means that the running
+      // trigger gets restarted.
+      for {
+        // Request a trigger be started for Alice.
+        resp <- startTrigger(uri, s"$testPkgId:TestTrigger:trigger", "Alice")
+        aliceTrigger <- parseTriggerId(resp)
+        triggerId <- parseTriggerId(resp)
+        // Proceed when it is confirmed to be running.
+        _ <- assertTriggerIds(uri, "Alice", _ == Vector(aliceTrigger))
+        // Simulate brief network connectivity loss. This will cause the
+        // running trigger's flow graph to complete with failure. Don't
+        // wait around to restore the network or the restart strategy
+        // will in turn lead to the stop strategy killing the trigger
+        // due to the lack of ability to initialize the restarted
+        // trigger.
+        _ <- Future { ledgerProxy.disable() }
+        _ <- Future { ledgerProxy.enable() }
+        // In the future I hope to be able to prove right here that the
+        // running trigger was failed and restarted. For now, the logs
```

Comment:
Agreed, maybe your trigger history map will help with this?

Reply:
I hope so!